core/arch/proc_api.h

```diff
@@ -167,24 +167,25 @@ typedef struct {
  */
 #ifdef AARCHXX
 
-typedef enum {
-    AA64ISAR0,
-    AA64ISAR1,
-    AA64PFR0,
-    AA64MMFR1,
-    AA64DFR0,
-    AA64ZFR0,
-    AA64PFR1,
-    AA64ISAR2,
-    AA64_NUM_FEAT_REGS
-} feature_reg_idx_t;
-
 /**
- * For AArch64 this struct holds features registers' values read by MRS instructions.
- * Used by proc_get_all_feature_bits().
+ * For AArch64 the isa_features array holds features registers' values read by MRS
+ * instructions. Used by proc_has_feature().
```

Comment:
I don't see this used by proc_has_feature().  In fact I don't see this used in any user-facing interface: should this be public at all?  Probably this should be moved to a non-public header.

Reply:
The AArch64 version of proc_has_feature() calls proc_has_feature_imp() which calls get_reg_val() which reads cpu_info.features.isa_features

I guess I was copying the style of the X86 comment above.